test/dataframe.jl

```diff
@@ -439,12 +439,40 @@ module TestDataFrame
         @test isa(df[1], Vector{Union{Int, Missing}})
         @test !isa(df[2], Vector{Union{Int, Missing}})
 
-        df = DataFrame(Any[collect(1:10), collect(1:10)])        
+        df = DataFrame(Any[collect(1:10), collect(1:10)])
         allowmissing!(df, [1,2])
         @test isa(df[1], Vector{Union{Int, Missing}}) && isa(df[2], Vector{Union{Int, Missing}})
 
-        df = DataFrame(Any[collect(1:10), collect(1:10)])        
+        df = DataFrame(Any[collect(1:10), collect(1:10)])
         allowmissing!(df)
         @test isa(df[1], Vector{Union{Int, Missing}}) && isa(df[2], Vector{Union{Int, Missing}})
+
+        df = DataFrame(Any[CategoricalArray(1:10),
+                           CategoricalArray(string.('a':'j'))])
+        allowmissing!(df)
+        @test all(issubtype.(typeof.(df.columns), CategoricalVector))
+        @test eltypes(df)[1] <: Union{CategoricalValue{Int}, Missing}
+        @test eltypes(df)[2] <: Union{CategoricalString, Missing}
+    end
+
+    @testset "similar" begin
+        df = DataFrame(a = ["foo"],
+                       b = CategoricalArray(["foo"]),
+                       c = [0.0],
+                       d = CategoricalArray([0.0]))
+        @test typeof.(df.columns) == typeof.(similar(df).columns)
```

Comment:
Also test the size of the result, and passing a valid number or rows different from the original?